### PR TITLE
sql: add ddl section `create index`

### PR DESCRIPTION
**Pull request by jackysp**
Same as title.

### CODE REVIEW

**Comment by shenli** [comment]
@sunhao2017

File: sql/ddl.md [hunk 1]
```diff
@@ -226,6 +226,52 @@ RENAME TABLE
...
+index_option:
+    KEY_BLOCK_SIZE [=] value
+  | index_type
+  | COMMENT 'string'
+
+index_type:
+    USING {BTREE | HASH}
+```
+
+[CREATE INDEX](#create-index-语法) 语句的作用是为一个已经存在的表创建一个索引。在功能上，[CREATE INDEX](#create-index-语法) 对应于 [ALTER TABLE](#alter-table-语法) 语句的创建索引功能。与 MySQL 一样 [CREATE INDEX](#create-index-语法) 不能创建主键索引。
```
Comment (andelf) [inline sql/ddl.md]: Bare Backquote for code formatting is enough.
Is full level of cross reference required? 
If so, markdown shouldn't be used

File: sql/ddl.md [hunk 2]
```diff
@@ -226,6 +226,52 @@ RENAME TABLE
...
+
+### 在语法上：
+
+* 支持 *UNIQUE* 索引，不支持 *FULLTEXT* 和 *SPATIAL* 索引。
```
Comment (andelf) [inline sql/ddl.md]: Do not mix coding style and emphasized style. Btw, we ignores emphasized style in PDF.

Use backquotes

File: sql/ddl.md [hunk 3]
```diff
@@ -226,6 +226,52 @@ RENAME TABLE
...
+
+* ***index_col_name*** 支持长度选项，以及索引排序选项 *ASC* 和 *DESC* 。
```
Comment (andelf) [inline sql/ddl.md]: There is no style uses 3 '*'s

File: sql/ddl.md (hunk 1 above)
Comment (jackysp) [reply sql/ddl.md]: It is the same style as MySQL create index document.

File: sql/ddl.md (hunk 2 above)
Comment (jackysp) [reply sql/ddl.md]: ditto

File: sql/ddl.md (hunk 3 above)
Comment (jackysp) [reply sql/ddl.md]: ditto

Comment (andelf) [reply sql/ddl.md]: ditto.........

File: sql/ddl.md [hunk 4]
```diff
@@ -226,6 +226,52 @@ RENAME TABLE
...
+
+* `index_col_name` 支持长度选项，以及索引排序选项 `ASC` 和 `DESC` 。
+
+* `index_option` 支持 `KEY_BLOCK_SIZE` 、`index_type` 和 `COMMENT`，不支持 `WITH PARSER` 。
+
+* `index_type` 支持 `BTREE` 和 `HASH` 。
+
+* 不支持 MySQL 的 `algorithm_option` 和 `lock_option` 选项。
+
+### 在功能上：
```
Comment (shenli) [inline sql/ddl.md]: Please reorg this section as we discussed.

**Comment by jackysp** [comment]
PTAL @shenli

**Pull request merged**